libs/ui/components/src/lib/molecules/profilePreview/ProfilePreview.tsx

```diff
@@ -0,0 +1,215 @@
+import { PublicUserBase } from '@tw/data';
+import { InvalidationData, linksRecords } from '@tw/ui/common';
+import { ConnectButton, SecondaryButton } from '@tw/ui/components';
+import { usePublicUserSocialStatsQuery } from '@tw/ui/data-access';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+type ProfilePreviewProps = {
+  displayedUser: PublicUserBase;
+  meId: string;
+  publicUserId?: string;
+  showConnectButton?: boolean;
+  invData: InvalidationData;
+};
+
+export function ProfilePreview({
+  displayedUser,
+  meId,
+  publicUserId,
+  showConnectButton,
+  invData,
+}: ProfilePreviewProps) {
+  const { id, name, avatar, uniqueName, bio, followingStatus } = displayedUser;
+  const { data: socialStats } = usePublicUserSocialStatsQuery(id);
+  const navigate = useNavigate();
+
+  const handleAvatarClick = () => {
+    navigate(linksRecords.publicProfilePage.baseById(id));
+  };
+
+  const followingCount = socialStats?.followingCount ?? 0;
+  const followersCount = socialStats?.followersCount ?? 0;
+
+  const connectButton = showConnectButton ? (
+    <ConnectButton
+      meId={meId}
+      buttonRelatedUserId={id}
+      publicUserId={publicUserId}
+      followingStatus={followingStatus}
+      invData={invData}
+    />
+  ) : null;
+
+  return (
+    <Popup>
+      <Header>
+        <ProfileImage $backgroundImage={avatar} onClick={handleAvatarClick} />
+        {connectButton}
+      </Header>
+
+      <ProfileDetailsWrapper>
+        <NameWrapper>
+          {/* //truncate */}
+          <NameLink onClick={handleAvatarClick}>{name}</NameLink>
+          <Handle>{uniqueName}</Handle>
+        </NameWrapper>
+        <Bio>{bio}</Bio>
+        <Stats>
+          <div>
+            <Count>{followingCount}</Count> <Label>Following</Label>
+          </div>
+          <div>
+            <Count>{followersCount}</Count> <Label>Followers</Label>
+          </div>
+        </Stats>
+
+        <SecondaryButton>Profile Summary</SecondaryButton>
+      </ProfileDetailsWrapper>
+    </Popup>
+  );
+}
+
+const Popup = styled.div(({ theme }) => {
+  const { black, white } = theme.colors;
+  const { 0.75: mt, 1.5: pY, 1: pX } = theme.spacing;
+  const { lg } = theme.radii;
```

Comment:
look at this part for example, it doesnt mean anything just extra syntax.

code has to be maintainable and simple